micrometer-core/src/test/java/io/micrometer/core/instrument/binder/cache/CaffeineCacheMetricsTest.java

```diff
@@ -0,0 +1,123 @@
+import com.github.benmanes.caffeine.cache.CacheLoader;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import com.github.benmanes.caffeine.cache.stats.CacheStats;
+
+import io.micrometer.core.instrument.FunctionCounter;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.TimeGauge;
+import io.micrometer.core.instrument.search.MeterNotFoundException;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+/**
+ * Tests for {@link CaffeineCacheMetrics}.
+ *
+ * @author Oleksii Bondar
+ */
+class CaffeineCacheMetricsTest extends AbstractCacheMetricsTest {
+
+    private Tags expectedTag = Tags.of("app", "test");
+    private LoadingCache<String, String> cache = Caffeine.newBuilder().build(new CacheLoader<String, String>() {
+        public String load(String key) throws Exception {
+            return "";
+        };
+    });
+    private CaffeineCacheMetrics metrics = new CaffeineCacheMetrics(cache, "testCache", expectedTag);
+
+    @Test
+    void reportExpectedGeneralMetrics() {
+        SimpleMeterRegistry registry = new SimpleMeterRegistry();
+        metrics.bindTo(registry);
+
+        verifyCommonCacheMetrics(registry);
+
+        Gauge evictionWeight = fetch(registry, "cache.eviction.weight").gauge();
+        CacheStats stats = cache.stats();
+        assertThat(evictionWeight.value()).isEqualTo(stats.evictionCount());
+
+        // specific to LoadingCache instance
+        TimeGauge loadDuration = fetch(registry, "cache.load.duration").timeGauge();
+        assertThat(loadDuration.value()).isEqualTo(stats.totalLoadTime());
+
+        FunctionCounter successfulLoad = fetch(registry, "cache.load", Tags.of("result", "success")).functionCounter();
+        assertThat(successfulLoad.count()).isEqualTo(stats.loadSuccessCount());
+
+        FunctionCounter failedLoad = fetch(registry, "cache.load", Tags.of("result", "failure")).functionCounter();
+        assertThat(failedLoad.count()).isEqualTo(stats.loadFailureCount());
+    }
+
+    @Test
+    void constructInstanceViaStaticMethodMonitor() {
+        SimpleMeterRegistry meterRegistry = new SimpleMeterRegistry();
+        CaffeineCacheMetrics.monitor(meterRegistry, cache, "testCache", expectedTag);
+
+        meterRegistry.get("cache.eviction.weight").tags(expectedTag).gauge();
+    }
+
+    @Test
+    void doNotReportMetricsForNonLoadingCache() {
+        SimpleMeterRegistry meterRegistry = new SimpleMeterRegistry();
+        Cache<Object, Object> cache = Caffeine.newBuilder().build();
+        CaffeineCacheMetrics metrics = new CaffeineCacheMetrics(cache, "testCache", expectedTag);
+        metrics.bindTo(meterRegistry);
+
+        assertThrows(MeterNotFoundException.class, () -> {
```

Comment:
You can use `assertThat(meterRegistry.find("cache.load.duration").timeGauge()).isNull())`, which is a bit simpler. `find` is a relaxed version of the stricter `get`. Since the meter won't exist at all, I think it's unnecessary to add the `expectedTag` to the search.

Reply:
Thanks, addressed